src/Dto/Segments/CreateSegment.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segments;
+
+use OneSignal\Dto\AbstractDto;
+use OneSignal\Dto\Filters\FilterAmountSpent;
+use OneSignal\Dto\Filters\FilterAppVersion;
+use OneSignal\Dto\Filters\FilterBoughtSku;
+use OneSignal\Dto\Filters\FilterConditional;
+use OneSignal\Dto\Filters\FilterCountry;
+use OneSignal\Dto\Filters\FilterFirstSession;
+use OneSignal\Dto\Filters\FilterLanguage;
+use OneSignal\Dto\Filters\FilterLastSession;
+use OneSignal\Dto\Filters\FilterLocation;
+use OneSignal\Dto\Filters\FilterSessionCount;
+use OneSignal\Dto\Filters\FilterSessionTime;
+use OneSignal\Dto\Filters\FilterTag;
+
+class CreateSegment implements AbstractDto
+{
+    /**
+     * @var non-empty-string
+     */
+    protected string $name;
+
+    /**
+     * @var array<int, FilterAmountSpent|FilterAppVersion|FilterBoughtSku|FilterConditional|FilterCountry|FilterFirstSession|FilterLanguage|FilterLastSession|FilterLocation|FilterSessionCount|FilterSessionTime|FilterTag>
```

Comment:
Hmm.. This union is so long :sweat_smile: Maybe better would be introduce some sort of marker interface like `FilterInterface` which would just extend AbstractDTO, and allow for any filter then, in case OneSignal would add more of them,so people could add these in their apps and use them